tests/main/snapd-apparmor/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: Ensure snapd-apparmor works as expected
+
+environment:
+    CONSUMER_SNAP: test-snapd-policy-app-consumer
+
+debug: |
+    journalctl -u snap.apparmor.service
+
+execute: |
+    if ! systemctl is-active snapd.apparmor.service; then
+      echo "Skipping test since snapd.apparmor.service is not active"
+      exit 0
+    fi
+
+    echo "Ensure snapd.apparmor is enabled"
+    systemctl is-enabled snapd.apparmor.service
+
+    # install a test snap which generates a lot of apparmor policies
+    echo "Given a test snap is installed"
+    "$TESTSTOOLS"/snaps-state install-local "$CONSUMER_SNAP"
+    tests.cleanup defer snap remove --purge "$CONSUMER_SNAP"
+
+    # stop snapd so it does not try and load apparmor policies directly
+    echo "And snapd is stopped"
+    systemctl stop snapd.service
+    tests.cleanup defer systemctl restart snapd.service
+
+    # get the current set of profiles but ignore any which may have gotten
+    # created on-the-fly by apparmor for namespaced commands - these
+    # contain / - as these won't get automatically recreated by
+    # snapd.apparmor
+    grep -v / /sys/kernel/security/apparmor/profiles | cut -f1 -d" " | sort > profiles.txt
+
+    # manually unload all profiles defined by snap apparmor policies
+    echo "If we unload existing snap apparmor policy"
+    for p in /var/lib/snapd/apparmor/profiles/*; do
+      while IFS= read -r profile; do
+        echo "Unloading $profile..."
+        echo -n "$profile" > /sys/kernel/security/apparmor/.remove
+        # check it is now unloaded
+        NOMATCH "^$profile " /sys/kernel/security/apparmor/profiles
```

Comment:
If the space after `$profile` is necessary, it might be better to explain it in the comment.

Reply:
Thanks - done in https://github.com/snapcore/snapd/pull/11129/commits/1c11c73422d42ada8e0dd4bb646089f3614af516